packages/core/src/context/experiment/message-context.tsx

```diff
@@ -27,9 +28,10 @@ export type Message = {
 const MessageContext = createContext<
   | {
       messages: Map<string, Message[]>
-      setMessage: (id: string, message: Message) => void
+      setMessage: (key: string, message: Message) => void
       setMessages: (messages: Map<string, Message[]>) => void
-      clearMessages: () => void
+      clearAllMessages: () => void
+      clearMessagesWithId: (messageId: string) => void
```

Comment:
Will clearMessagesWithId be able to hit more than one message? If not it could be called clearMessage(messageId: string)

Reply:
The idea is to tag and target multiple messages. So e.g. all "validation-error" messages can be cleared at once.